src/main/java/test/project/bookshop/validation/FieldMatchValidator.java

```diff
@@ -14,7 +16,14 @@ public void initialize(final FieldMatch constraintAnnotation) {
     }
 
     @Override
-    public boolean isValid(String s, ConstraintValidatorContext constraintValidatorContext) {
-        return firstFieldName != null && firstFieldName.equals(secondFieldName);
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        try {
```

Comment:
You can use 
`Object field = new BeanWrapperImpl(value).getPropertyValue(this.firstFieldName);`
and 
`return Objects.equals(field, fieldMatch);`